shell/platform/common/public/flutter_texture_registrar.h

```diff
@@ -36,6 +58,26 @@ typedef struct {
   size_t height;
 } FlutterDesktopPixelBuffer;
 
+// A GPU surface descriptor.
+typedef struct {
+  // The surface handle.
+  // For DirectX textures (kFlutterDesktopGpuSurfaceTypeDxgi), this is the
+  // shared handle of an IDXGIResource.
+  void* handle;
```

Comment:
What is the ownership policy here? Does engine take over the reference, or does it increases the reference count?

Reply:
I guess the surface descriptor being passed as const sort of answers my questions.

Reply:
ANGLE increments the reference count once the shared handle has been opened.

However, as this happens asynchronously due to the GPU surface descriptor callback, it's advised to explicitly increment the texture's reference count before the callback returns in order to prevent potential races (a plugin might destroy the texture after the callback returns but just before the Flutter texture gets populated). This obviously requires a `FlutterDesktopGpuSurfaceDescriptor::release_callback` keeping the reference count in balance.